Canine.py

```diff
@@ -3,8 +3,9 @@
 
 class Canine(Animal, metaclass=ABCMeta):
     @abstractmethod
-    def __init__(self, name):
+    def __init__(self, name, makeNoise=None):
         super().__init__(name)
 
+
```

Comment:
Please define _makeNoise(self): here
It is because of reducing duplicate code (in Wold and Dog)

    def _makeNoise(self):
        return super()._getName() + self._makeNoiseAlgo.makeNoise()